plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/meta/CubridMetaModel.java

```diff
@@ -0,0 +1,120 @@
+
+public class CubridMetaModel extends GenericMetaModel {
+
+	private static final Log log = Log.getLog(CubridMetaModel.class);
+
+	public CubridMetaModel() {
+	}
+
+	public boolean isSupportMultiSchema(JDBCSession session) {
+		try {
+			int major = session.getMetaData().getDatabaseMajorVersion();
+			int minor = session.getMetaData().getDatabaseMinorVersion();
+			if (major > 11 || (major == 11 && minor >= 2)) {
+				return true;
+			}
+		} catch (SQLException e) {
+			log.error("Can't get database version", e);
+		}
+		return false;
+	}
+
+	@Override
+	public JDBCStatement prepareTableColumnLoadStatement(@NotNull JDBCSession session,
+			@NotNull GenericStructContainer owner, @Nullable GenericTableBase forTable) throws SQLException {
+		if (isSupportMultiSchema(session) && forTable instanceof CubridTable) {
+			CubridTable table = (CubridTable) forTable;
+			return session.getMetaData().getColumns(null, null, table != null ? table.getUniqueName() : null, null)
+					.getSourceStatement();
+		}
+		return super.prepareTableColumnLoadStatement(session, owner, forTable);
+	}
+
+	@Override
+	public JDBCStatement prepareUniqueConstraintsLoadStatement(@NotNull JDBCSession session,
+			@NotNull GenericStructContainer owner, @Nullable GenericTableBase forTable)
+			throws SQLException, DBException {
+		if (isSupportMultiSchema(session) && forTable instanceof CubridTable) {
+			CubridTable table = (CubridTable) forTable;
+			return session.getMetaData().getPrimaryKeys(null, null, table != null ? table.getUniqueName() : null)
+					.getSourceStatement();
+		}
+		return super.prepareUniqueConstraintsLoadStatement(session, owner, forTable);
+	}
+
+	@Override
+	public JDBCStatement prepareForeignKeysLoadStatement(@NotNull JDBCSession session,
+			@NotNull GenericStructContainer owner, @Nullable GenericTableBase forTable) throws SQLException {
+		if (isSupportMultiSchema(session) && forTable instanceof CubridTable) {
+			CubridTable table = (CubridTable) forTable;
+			return session.getMetaData().getImportedKeys(null, null, table != null ? table.getUniqueName() : null)
+					.getSourceStatement();
+		}
+		return super.prepareForeignKeysLoadStatement(session, owner, forTable);
+	}
+
+	public JDBCStatement prepareIndexLoadStatement(@NotNull JDBCSession session, CubridTable table)
+			throws SQLException {
+		String tableName = isSupportMultiSchema(session) ? table.getUniqueName() : table.getName();
+		return session.getMetaData().getIndexInfo(null, null, tableName, false, true).getSourceStatement();
+	}
+
+	public CubridTable createTableImpl(@NotNull JDBCSession session, @NotNull CubridObjectContainer owner,
+			@NotNull GenericMetaObject tableObject, @NotNull JDBCResultSet dbResult) {
+
+		String tableName = JDBCUtils.safeGetString(dbResult, CubridConstants.CLASS_NAME);
+		String tableType = JDBCUtils.safeGetStringTrimmed(dbResult, JDBCConstants.TABLE_TYPE);
+
+		CubridTable table = this.createTableImpl(owner, tableName, tableType, dbResult);

```

Comment:
You can pass session monitor to createTableImpl, so the function may know if loading was cancelled by user

Reply:
I have fixed it.